lms/djangoapps/courseware/management/commands/dump_to_neo4j.py

```diff
@@ -0,0 +1,211 @@
+"""
+This file contains a management command for exporting the modulestore to
+neo4j, a graph database.
+"""
+from __future__ import unicode_literals
+
+import logging
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+from django.utils import six
+from py2neo import Graph, Node, Relationship, authenticate
+from py2neo.compat import integer, string, unicode as neo4j_unicode
+from request_cache.middleware import RequestCache
+from xmodule.modulestore.django import modulestore
+
+log = logging.getLogger(__name__)
+
+# When testing locally, neo4j's bolt logger was noisy, so we'll only have it
+# emit logs if there's an error.
+bolt_log = logging.getLogger('neo4j.bolt')  # pylint: disable=invalid-name
+bolt_log.setLevel(logging.ERROR)
+
+ITERABLE_NEO4J_TYPES = (tuple, list, set, frozenset)
+PRIMITIVE_NEO4J_TYPES = (integer, string, neo4j_unicode, float, bool)
+
+
+class ModuleStoreSerializer(object):
+    """
+    Class with functionality to serialize a modulestore into subgraphs,
+    one graph per course.
+    """
+    def __init__(self):
+        self.all_courses = modulestore().get_course_summaries()
+
+    @staticmethod
+    def serialize_item(item):
+        """
+        Args:
+            item: an XBlock
+
+        Returns:
+            fields: a dictionary of an XBlock's field names and values
+            label: the name of the XBlock's type (i.e. 'course'
+            or 'problem')
+        """
+        # convert all fields to a dict and filter out parent and children field
+        fields = dict(
+            (field, field_value.read_from(item))
+            for (field, field_value) in six.iteritems(item.fields)
+            if field not in ['parent', 'children']
+        )
+
+        course_key = item.scope_ids.usage_id.course_key
+
+        # set or reset some defaults
+        fields['edited_on'] = six.text_type(getattr(item, 'edited_on', ''))
+        fields['display_name'] = item.display_name_with_default
+        fields['org'] = course_key.org
+        fields['course'] = course_key.course
+        fields['run'] = course_key.run
+        fields['course_key'] = six.text_type(course_key)
+
+        label = item.scope_ids.block_type
+
+        # prune some fields
+        if label == 'course':
+            if 'checklists' in fields:
+                del fields['checklists']
```

Comment:
@Qubad786 this field only shows up in courses, and I didn't think it would be useful to pass over to neo4j.